### PR TITLE
Indicator Records module + basic tests

### PR DESCRIPTION
**Pull request by alexeh**
This PR adds:

`Indicator Records` module and basic tests

**NOTE**
DTO, required fields and etc... been told me by a magician

### CODE REVIEW

File: api/src/modules/indicator-records/indicator-record.entity.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  @Column({
+    type: 'enum',
+    enum: TASK_STATUS,
+    default: TASK_STATUS.UNSTARTED,
+  })
+  status!: TASK_STATUS;
+
+  @ApiProperty()
+  @Column({ nullable: true })
+  statusMsg: string;
```
Comment (tiagojsag) [inline api/src/modules/indicator-records/indicator-record.entity.ts:56]: add the ?

File: api/src/modules/indicator-records/indicator-records.controller.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    const results = await this.indicatorRecordService.findAllPaginated(
+      fetchSpecification,
+    );
+    return this.indicatorRecordService.serialize(
+      results.data,
+      results.metadata,
+    );
+  }
+
+  @ApiOperation({ description: 'Find indicator resource by id' })
```
Comment (tiagojsag) [inline api/src/modules/indicator-records/indicator-records.controller.ts:73]: indicator resource => indicator record

File: api/src/modules/indicator-records/indicator-records.controller.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  @ApiOkResponse({ type: IndicatorRecord })
+  @JSONAPISingleEntityQueryParams()
+  @Get(':id')
+  async findOne(@Param('id') id: string): Promise<IndicatorRecord> {
+    return await this.indicatorRecordService.serialize(
+      await this.indicatorRecordService.getById(id),
+    );
+  }
+
+  @ApiOperation({ description: 'Create a indicator resource' })
```
Comment (tiagojsag) [inline api/src/modules/indicator-records/indicator-records.controller.ts:83]: same here and below

File: api/src/modules/indicators/indicator.repository.ts [hunk 4]
```diff
@@ -2,4 +2,11 @@ import { EntityRepository, Repository } from 'typeorm';
 import { Indicator } from 'modules/indicators/indicator.entity';
 
 @EntityRepository(Indicator)
-export class IndicatorRepository extends Repository<Indicator> {}
+export class IndicatorRepository extends Repository<Indicator> {
+  //TODO add response DTO
```
Comment (tiagojsag) [inline api/src/modules/indicators/indicator.repository.ts:6]: this seems to be unrelated. Also, you dont need a response DTO here

**Review by tiagojsag — CHANGES_REQUESTED** [review]

File: api/src/modules/indicators/indicator.repository.ts (hunk 4 above)
Comment (alexeh) [reply api/src/modules/indicators/indicator.repository.ts:6]: Yep, this went here by mistake while I was implementing the indicator-data feature. THX for pointing out

File: api/src/modules/indicator-records/indicator-record.entity.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  entitiesAllowedAsIncludes: ['indicatorRecords'],
+  columnsAllowedAsFilter: [
+    'value',
+    'status',
+    'sourcingRecordId',
+    'indicatorId',
+  ],
+};
+export enum TASK_STATUS {
```
Comment (tiagojsag) [inline api/src/modules/indicator-records/indicator-record.entity.ts:29]: rename this to something like INDICATOR_RECORD_STATUS

File: api/src/modules/indicator-records/indicator-records.service.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  ) {
+    super(
+      indicatorRecordRepository,
+      indicatorRecordResource.name.singular,
+      indicatorRecordResource.name.plural,
+    );
+  }
+  get serializerConfig(): JSONAPISerializerConfig<IndicatorRecord> {
+    return {
+      attributes: ['value', 'status', 'statusMsg', 'createdAt', 'updatedAt'],
```
Comment (tiagojsag) [inline api/src/modules/indicator-records/indicator-records.service.ts:35]: some attributes are missing here

**Review by tiagojsag — CHANGES_REQUESTED** [review]
